kythe/java/com/google/devtools/kythe/platform/java/JavacOptionsUtils.java

```diff
@@ -312,4 +342,12 @@ public static Charset getEncodingOption(List<String> options) {
     int i = options.lastIndexOf("-encoding");
     return (i >= 0) ? Charset.forName(options.get(i + 1)) : null;
   }
+
+  /** Returns a copy of iter, unless iter is zero length or null. */
```

Comment:
I don't think we need all these nulls.  We can just return an empty list or an empty Iterator.